silverback/subscriptions.py

```diff
@@ -46,14 +46,23 @@ async def __anext__(self) -> str:
         if not self.connection:
             raise StopAsyncIteration
 
-        message = await self.connection.recv()
+        return await self._receive()
+
+    async def _receive(self, timeout: Optional[int] = None) -> str:
+        """Receive (and wait if no timeout) for the next message from the
+        socket.
+        """
+        if not self.connection:
+            raise ConnectionError("Connection not opened")
+
+        message = await asyncio.wait_for(self.connection.recv(), timeout)
```

Comment:
Is this the location where node filter drops end up raising?

Reply:
Looks like it, according to the trace on #84

Reply:
Nice, we might want to come back later and handle this timeout by reconnecting